src/sentry/release_health/metrics.py

```diff
@@ -615,59 +634,49 @@ def check_has_health_data(
+        select = [MetricField(metric_mri=SessionMRI.SESSION.value, alias="value", op="sum")]
+
+        where_clause = []
+        groupby = [
+            MetricGroupByField(field="project_id"),
         ]
 
         if includes_releases:
-            releases = [x[1] for x in projects_list]  # type: ignore
-            release_column_name = resolve_tag_key(USE_CASE_ID, org_id, "release")
-            releases_ids = resolve_many_weak(USE_CASE_ID, org_id, releases)
-            where_clause.append(Condition(Column(release_column_name), Op.IN, releases_ids))
-            column_names = ["project_id", release_column_name]
-
-        else:
-            column_names = ["project_id"]
-
-        def extract_row_info_func(
-            include_releases: bool,
-        ) -> Callable[[Mapping[str, Union[int, str]]], ProjectOrRelease]:
-            def f(row: Mapping[str, Union[int, str]]) -> ProjectOrRelease:
-                if include_releases:
-                    return row["project_id"], reverse_resolve(USE_CASE_ID, org_id, row.get(release_column_name))  # type: ignore
-                else:
-                    return row["project_id"]  # type: ignore
-
-            return f
-
-        extract_row_info = extract_row_info_func(includes_releases)
-
-        query_cols = [Column(column_name) for column_name in column_names]
-        group_by_clause = query_cols
-
-        query = Query(
-            match=Entity(EntityKey.MetricsCounters.value),
-            select=query_cols,
+            where_clause.append(filter_releases_by_project_release(projects_list))  # type: ignore
+            groupby.append(MetricGroupByField(field="release"))
+
+        query = MetricsQuery(
+            org_id=org_id,
+            project_ids=project_ids,
+            select=select,
+            start=start,
+            end=now,
+            granularity=Granularity(DAY),
+            groupby=groupby,
             where=where_clause,
-            groupby=group_by_clause,
-            granularity=Granularity(24 * 60 * 60),  # daily
-        )
-        request = Request(dataset=Dataset.Metrics.value, app_id=SnubaAppID, query=query)
-        result = raw_snql_query(
-            request, referrer="release_health.metrics.check_has_health_data", use_cache=False
+            include_series=False,
+            include_totals=True,
         )
 
-        return {extract_row_info(row) for row in result["data"]}
+        raw_result = get_series(
+            projects=projects,
+            metrics_query=query,
+            use_case_id=USE_CASE_ID,
+        )
+        groups = raw_result["groups"]
+
+        ret_val = set()
+        for group in groups:
+            if includes_releases:
+                by = group.get("by", {})
+                idx = by.get("project_id"), by.get("release")
+                ret_val.add(idx)
+            else:
+                proj_id = get_path(group, "by", "project_id")
+                ret_val.add(proj_id)
+        return ret_val  # type: ignore
```

Comment:
`from typing import Set`
if you annotate `ret_val: Set` does it help to avoid type ignore?

Reply:
Checked,
Unfortunately no.
The issue is not that ret_val is a set ( it knows this), the issue is that it can be a Set[int] or a Set[Tuple[in,str]] depending on whether we return projectIds or project-release tuples ... 
Defining : `ret_val: Set` is an error since Set is a generic type so it needs an argument
Defining: `ret_val:Set[ProjectOrRelease]` also fails although strictly speaking this is the correct returned type.
For some reason if I define 
`ret_val: Set[ProjectOrRelease]` it generates an error in the if part, when I  add Project ids to the set which should be technically correct.
I didn't find a way to get it to pass mypy without an ignore stament.

Reply:
Okey, let's resolve it.

Reply:
I'm pretty sure that it would pass if I declared `ProjectOrRelease` as being
Union[ProjectId, ProjectRelase] but that is not really the correct type since containers of `ProjectOrRelease` should have either one or the other but not both... 
I think this is actually the problem of mypy, if sees that sometime I ad an `int` to the set and sometimes I add a `Tuple[int,str]` but it can't prove that I never add both in one invocation.